packages/react-pose/src/components/Transition/index.tsx

```diff
@@ -19,17 +20,39 @@ class Transition extends React.Component<Props, State> {
     props: Props,
     { isLeaving, removeFromTree, children }: State
   ) => {
-    const incomingChildren = makeChildList(props.children);
+    const {
+      children: groupChildren,
+      preEnterPose,
+      enterPose,
+      exitPose,
+      flipMove,
+      animateOnMount,
+      ...propsForChild,
+    } = props;
 
-    return {
+    const incomingChildren = makeChildList(groupChildren);
+
+    if (process.env.NODE_ENV !== 'production') {
+      warning(!propsForChild.onPoseComplete, '<Transition/> (or <PoseGroup/>) doesn\'t accept onPoseComplete prop.')
```

Comment:
Just a style thing but I'd prefer backticks here without string escaping. Also although we had a ticket about it I wonder about adding warnings about unsupported props. It's a rabbit hole that we could easily litter the codebase with and there was no documentation to suggest it is supported

Reply:
This is valid concern, I've added it for 2 reasons though
- you've tried to pass it in previous version of the fix [here](https://github.com/Popmotion/popmotion/pull/496/files#diff-d6d4f941ac5e823f38de0ba45db686a8R255) 😄 
- but on a serious note - `<Transition onPoseComplete={() => {}}>` would override child's `onPoseComplete` and while learning the API it might be an easy thing to miss, so to improve DX here I would advocate to warn about this case, especially that this is dev-only warning so won't cost anything in production builds

Reply:
Yeah makes sense, good thinking